core/src/main/java/org/everit/json/schema/ObjectComparator.java

```diff
@@ -1,20 +1,32 @@
 package org.everit.json.schema;
 
-import static org.everit.json.schema.loader.OrgJsonUtil.getNames;
+import org.json.JSONArray;
+import org.json.JSONObject;
 
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.text.NumberFormat;
 import java.util.Arrays;
 import java.util.Objects;
 
-import org.json.JSONArray;
-import org.json.JSONObject;
+import static org.everit.json.schema.loader.OrgJsonUtil.getNames;
+
 
 /**
  * Deep-equals implementation on primitive wrappers, {@link JSONObject} and {@link JSONArray}.
  */
 public final class ObjectComparator {
 
+    private static NumberFormat numberFormat = NumberFormat.getInstance();
+
+    static {
+        numberFormat.setMaximumFractionDigits(Integer.MAX_VALUE);
+        numberFormat.setGroupingUsed(false);
+    }
```

Comment:
Format classes are not thread safe. Please let's create a separate instance whenever it is used (so a `private static NumberFormat createNumberFormat()` method).